controllers/secretsync_controller.go

```diff
@@ -0,0 +1,469 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	secretsyncv1alpha1 "sigs.k8s.io/secret-sync-controller/api/v1alpha1"
+	"sigs.k8s.io/secret-sync-controller/pkg/k8s"
+	"sigs.k8s.io/secret-sync-controller/pkg/provider"
+	"sigs.k8s.io/secret-sync-controller/pkg/util/secretutil"
+	secretsstorecsiv1 "sigs.k8s.io/secrets-store-csi-driver/apis/v1"
+	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
+)
+
+const (
+	// CSIPodName is the name of the pod that the mount is created for
+	CSIPodName = "csi.storage.k8s.io/pod.name"
+
+	// CSIPodNamespace is the namespace of the pod that the mount is created for
+	CSIPodNamespace = "csi.storage.k8s.io/pod.namespace"
+
+	// CSIPodUID is the UID of the pod that the mount is created for
+	CSIPodUID = "csi.storage.k8s.io/pod.uid"
+
+	// CSIPodServiceAccountName is the name of the pod service account that the mount is created for
+	CSIPodServiceAccountName = "csi.storage.k8s.io/serviceAccount.name"
+
+	// CSIPodServiceAccountTokens is the service account tokens of the pod that the mount is created for
+	CSIPodServiceAccountTokens = "csi.storage.k8s.io/serviceAccount.tokens" //nolint
+
+	// Label applied by the controller to the secret object
+	ControllerLabelKey = "secrets-store.sync.x-k8s.io"
+
+	// Label applied by the controller to the secret object
+	ControllerAnnotationKey = "secrets-store.sync.x-k8s.io"
+
+	// Version is the version of the secret sync controller
+	Version = "v1"
+
+	// SecretSyncControllerFieldManager is the field manager used by the secret sync controller
+	SecretSyncControllerFieldManager = Version + "-secret-sync-controller"
+
+	// Environment variables set using downward API to pass as params to the controller
+	// Used to maintain the same logic as the Secrets Store CSI driver
+	SyncControllerPodName = "SYNC_CONTROLLER_POD_NAME"
+	SyncControllerPodUID  = "SYNC_CONTROLLER_POD_UID"
+)
+
+type AllClientBuilder interface {
+	Get(ctx context.Context, provider string) (v1alpha1.CSIDriverProviderClient, error)
+}
+
+// SecretSyncReconciler reconciles a SecretSync object
+type SecretSyncReconciler struct {
+	client.Client
+	Audiences            []string
+	Clientset            *kubernetes.Clientset
+	Scheme               *runtime.Scheme
+	TokenClient          *k8s.TokenClient
+	ProviderClients      AllClientBuilder
+	RotationPollInterval time.Duration
+	EventRecorder        record.EventRecorder
+}
+
+//+kubebuilder:rbac:groups=secret-sync.x-k8s.io,resources=secretsyncs,verbs=get;list;watch
+//+kubebuilder:rbac:groups=secret-sync.x-k8s.io,resources=secretsyncs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=create;patch;delete
```

Comment:
I don't think the `delete` verb is required, could you validate by removing it?

Reply:
I'll address this in the follow-up PR. Opened https://github.com/kubernetes-sigs/secrets-store-sync-controller/issues/16 for tracking.